google-cloud-tools-plugin/src/com/google/gct/idea/appengine/cloud/AppEngineDeploymentRunConfigurationEditor.java

```diff
@@ -236,20 +243,35 @@ protected JComponent createEditor() {
     private final String fileName;
     private final TextFieldWithBrowseButton filePicker;
     private final Supplier<File> sourceFileProvider;
+    private final JPanel fileSelector;
 
     public GenerateConfigActionListener(
         Project project,
         String fileName,
         Supplier<File> sourceFileProvider,
-        TextFieldWithBrowseButton filePicker) {
+        TextFieldWithBrowseButton filePicker,
+        JPanel fileSelector) {
       this.project = project;
       this.fileName = fileName;
       this.sourceFileProvider = sourceFileProvider;
       this.filePicker = filePicker;
+      this.fileSelector = fileSelector;
     }
 
     @Override
     public void actionPerformed(ActionEvent event) {
+      // This can happen only when the deployment source is UserSpecifiedPathDeploymentSource
+      // and the selected file does not have .jar or .war extension. Artifact deployment sources
+      // always have .jar or .war extensions because we filter out others.
+      if (sourceFileProvider.get() == null) {
```

Comment:
Cannot call that method directly inside this class, unfortunately. I think it's also a bit better to always return when it's null, although we know it cannot happen if it's not UserSpecifiedpathDeploymentSource.